package/ota_tool/src/main.rs

```diff
@@ -0,0 +1,160 @@
+#[macro_use]
+extern crate hyper;
+extern crate reqwest;
+extern crate uuid;
+#[macro_use]
+extern crate serde_derive;
+extern crate clap;
+
+use uuid::Uuid;
+use std::process::Command;
+use std::io;
+use std::io::{BufReader, Write};
+use std::fs::File;
+use std::path::Path;
+use std::iter::Iterator;
+use clap::{Arg, App};
+
+header! { (DeviceUid, "Device-Uid") => [Uuid] }
+header! { (CurrentVersion, "Current-Version") => [String] }
+
+#[derive(Deserialize, Debug)]
+struct OTAServiceResponse {
+    version: String,
+    sha256: String,
+    url: String,
+}
+
+fn sha256(path: &std::path::Path) -> Result<String, String> {
+    // Linux
+    let mut base_cmd = Command::new("sha256sum");
```

Comment:
going to stick with `sha256sum` to avoid pulling another package into buildroot. The OS X use case was really just for testing locally and shouldn't be needed in production.